src/main/java/duke/commands/CommandResult.java

```diff
@@ -0,0 +1,76 @@
+package duke.commands;
+
+import duke.tasks.TaskList;
+
+/**
+ * Represents the result of a command after execution.
+ */
+public class CommandResult {
+    private final String messageForUser;
+    private final TaskList updatedTaskList;
+    private final boolean isExiting;
```

Comment:
It might be best to capitalise the names of constants even if they are private attributes. I noticed this in several other files too so it might be good to do a quick check on this :) For your convenience, the bottom of this [quick guide](https://github.com/nus-cs2103-AY2021S2/forum/issues/93) shares how the `checkstyle.xml` can be updated to catch these violations :)

Reply:
Unless the application is behaving incorrectly with the final keyword, I am inclined to think that capitalizing the instance attributes would be the better option. The final keyword reliably ensures that the state of the attribute is never modified and this seems especially important for the taskList, which should not have a new list object unintentionally overwriting it. I have to admit I am not too experienced with java and have only read up on the use of final recently so I am open to and would appreciate any advice on this as well :D

Reply:
Shall ask the forum 😃